channeldb/migration30/migration.go

```diff
@@ -553,6 +553,11 @@ func convertRevocationLog(commit *mig.ChannelCommitment,
 		HTLCEntries:      make([]*HTLCEntry, 0, len(commit.Htlcs)),
 	}
 
+	if !noAmtData {
```

Comment:
Shouldn't this be a _new_ migration? There're already nodes that have run this prior migration to convert the revocation logs.

Reply:
there is no actual migration required for this new code. What this is doing is preventing the loss of this data for people who have not yet run this migration. So this is only useful for people who have not run the migration yet. 
Basically allows them to get the benefit of the main migration while not giving them a disadvantage when it comes to tower backups. 

Since the main migration is optional as well, I think there will be enough users who have not yet run the migration but will still want to one day. So this is to help those users out. This is also the main reason for pushing to get this in to 0.16

Reply:
(see test matrix above for what this means for various users)